crates/shared/src/http_solver.rs

```diff
@@ -99,8 +99,7 @@ impl HttpSolverApi for DefaultHttpSolverApi {
             .checked_sub(Duration::from_secs(1))
             .ok_or_else(|| anyhow!("no time left to send request"))?;
 
-        let mut url = self.base.clone();
-        url.set_path("/solve");
+        let mut url = self.base.clone().join("solve")?;
```

Comment:
nit: `Url::join` already takes a reference and returns a new `Url` instance, so `.clone()` here is superfluous.

```suggestion
        let mut url = self.base.join("solve")?;
```

I tried this out in [the Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=87cdbc6b4ed4ad09bad989780ef18d49) and it seems to work.